ignite/handlers/state_param_scheduler.py

```diff
@@ -0,0 +1,342 @@
+        super(LambdaStatePScheduler, self).__init__(param_name, save_history)
+        self.lambda_fn = lambda_fn
+        self._state_attrs += ["lambda_fn"]
+
+    def get_param(self) -> Union[List[float], float]:
+        return self.lambda_fn(self.event_index)
+
+
+class LinearStatePScheduler(StatePScheduler):
+    """Update a parameter during training by using linear function.
+    The function keeps the parameter value to zero until step_zero steps passed and then linearly increases it to 1
+    until an additional step_one steps passed. Continues the trend until it reaches max_value.
+
+    Args:
+        initial_value : starting value of the parameter.
+        step_constant : step index until parameter's value is kept constant.
+        step_max_value : step index at which parameter's value becomes max_value.
+        max_value : max parameter value.
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    Examples:
+
+        .. code-block:: python
+
+            ...
+            engine = Engine(train_step)
+
+            linear_state_parameter_scheduler = LinearStatePScheduler(
+                param_name="linear_scheduled_param", initial_value=0, step_constant=2, step_max_value=5, max_value=10,
+            )
+
+            linear_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+            engine.run([0] * 8, max_epochs=10)
+
+    .. versionadded:: 0.6.0
+    """
+
+    def __init__(
+        self,
+        initial_value: float,
+        step_constant: int,
+        step_max_value: int,
+        max_value: float,
+        param_name: str,
+        save_history: bool = False,
+    ):
+        super(LinearStatePScheduler, self).__init__(param_name, save_history)
+        self.initial_value = initial_value
+        self.step_constant = step_constant
+        self.step_max_value = step_max_value
+        self.max_value = max_value
+        self._state_attrs += ["initial_value", "step_constant", "step_max_value", "max_value"]
+
+    def get_param(self) -> Union[List[float], float]:
+        if self.event_index <= self.step_constant:
+            delta = 0.0
+        elif self.event_index > self.step_max_value:
+            delta = self.max_value - self.initial_value
+        else:
+            delta = (
+                (self.max_value - self.initial_value)
+                / (self.step_max_value - self.step_constant)
+                * (self.event_index - self.step_constant)
+            )
+
+        return self.initial_value + delta
+
+
+class ExpStatePScheduler(StatePScheduler):
+    """Update a parameter during training by using exponential function.
+    The function decays the parameter value by gamma every step.
+    Based on the closed form of ExponentialLR from Pytorch
+    https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L457
```

Comment:
Let's use links on docs instead of unfixed links on some line of code. Here and below